### DISCUSSION TITLE
Export `core` explicitly in `package.json`

### DISCUSSION THREAD

**Discussion by shishkin** [aws/aws-cdk#22912] [General]
I think the [suggested solution](https://github.com/aws/aws-cdk/issues/17950#issuecomment-991189490) to use the default export is not satisfactory. I'm running into issues when bundlers scan dependencies and choke on things inside the `aws-cdk-lib` that I don't even use (e.g. Dockerfile from `aws-cdk-lib/lambda-layer-kubectl/layer/Dockerfile` 🤷‍♂️ ), but they're still reachable from the bundler's perspective because my code imports `aws-cdk-lib` default export which exports *everything*.

Exporting `core` explicitly would allow for better tree shaking in bundlers. Otherwise `aws-cdk-lib` objectively still remains a monolithic CJS package and neglects benefits of ESM.

_Originally posted by @shishkin in https://github.com/aws/aws-cdk/issues/17950#issuecomment-1312691102_

**Comment by peterwoodworth** [comment]
We cannot change this until the next major version without it being a breaking change, so I think you'll need to find some sort of workaround for your problem. I'm converting this to a discussion for better longevity and searchability on the issue 🙂

**Comment by shishkin** [comment]
For anyone facing the same issue, I found a workaround by structuring exports of *my* package. I made sure that code importing CDK lib is exported via a separate export in `package.json` and is not imported anywhere from the main `index.ts` export.

**Comment by xChickens** [comment]
I don't understand - why would this be a breaking change? Why would adding a line for "./core" pointing to "./core/index.js" break things? The root (".") still points to "./index.js" so people would still be able to use the default export.

**Reply by xChickens** [reply]
@peterwoodworth
